src/components/NavBar.js

```diff
@@ -1,37 +1,63 @@
-import React from 'react';
+import React, {useContext} from 'react';
+import {Link, useNavigate} from 'react-router-dom';
 import logo from '../assets/banana-01.png';
-import { useHistory, Link } from 'react-router-dom';
+
+import {AuthContext} from "../context/AuthProvider";
 
 function NavBar() {
-  const history = useHistory();
+    const navigate = useNavigate();
+    const { isAuth, logout } = useContext(AuthContext)
 
-  return (
-    <nav>
-        <Link to="/">
-          <span className="logo-container">
-            <img src={logo} alt="logo"/>
-            <h3>
-              Banana Security
-            </h3>
-          </span>
-        </Link>
+    return (
+        <nav>
+            <Link to="/">
+                <span className="logo-container">
+                    <img src={logo} alt="logo"/>
+                    <h3>
+                        Banana Security
+                    </h3>
+                </span>
+            </Link>
 
-      <div>
-        <button
-          type="button"
-          onClick={() => history.push('/signin')}
-        >
-          Log in
-        </button>
-        <button
-          type="button"
-          onClick={() => history.push('/signup')}
-        >
-          Registreren
-        </button>
-      </div>
-    </nav>
-  );
+            { isAuth.auth &&
+            <div>
+                {/*These buttons are visible when authorized*/}
+                <button
+                    type="button"
+                    onClick={() => { navigate('/Profile') }}
```

Comment:
Als de functiebody maar één regel heeft, dan kun je de accolades weglaten
```suggestion
                    onClick={() => navigate('/Profile')}
```